### ISSUE TITLE
Getting ModuleNotFoundError: No module named 'numpy._core'

### ISSUE THREAD

**Issue by 1skol1** [kserve/kserve#3968]
/kind bug

**What steps did you take and what happened:**
Trying to replicate this guide of deploying sklearn model [deploy-the-model-with-rest-endpoint-through-inferenceservice](https://kserve.github.io/website/latest/modelserving/v1beta1/sklearn/v2/#deploy-the-model-with-rest-endpoint-through-inferenceservice)

**What did you expect to happen:**
Model container to run successfully 

**What's the InferenceService yaml:**
Output of  `kubectl get isvc $name -n $namespace -oyaml`
```
apiVersion: serving.kserve.io/v1beta1
kind: InferenceService
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"serving.kserve.io/v1beta1","kind":"InferenceService","metadata":{"annotations":{},"name":"sklearn-v2-iris","namespace":"kserve-models"},"spec":{"predictor":{"model":{"modelFormat":{"name":"sklearn"},"protocolVersion":"v2","runtime":"kserve-sklearnserver","storageUri":"s3://<bucket-name>/kserve-models/model.joblib"},"serviceAccountName":"kserve-models-sa"}}}
  creationTimestamp: "2024-09-30T13:11:57Z"
  finalizers:
  - inferenceservice.finalizers
  generation: 1
  name: sklearn-v2-iris
  namespace: kserve-models
  resourceVersion: "631220514"
  uid: 5a69406e-d235-4d1a-8034-b3713e0a49a2
spec:
  predictor:
    model:
      modelFormat:
        name: sklearn
      name: ""
      protocolVersion: v2
      resources: {}
      runtime: kserve-sklearnserver
      storageUri: s3://<bucket-name>/kserve-models/model.joblib
    serviceAccountName: kserve-models-sa
status:
  components:
    predictor:
      latestCreatedRevision: sklearn-v2-iris-predictor-00001
  conditions:
  - lastTransitionTime: "2024-09-30T13:11:57Z"
    reason: PredictorConfigurationReady not ready
    severity: Info
    status: Unknown
    type: LatestDeploymentReady
  - lastTransitionTime: "2024-09-30T13:11:57Z"
    severity: Info
    status: Unknown
    type: PredictorConfigurationReady
  - lastTransitionTime: "2024-09-30T13:11:57Z"
    message: Configuration "sklearn-v2-iris-predictor" is waiting for a Revision to
      become ready.
    reason: RevisionMissing
    status: Unknown
    type: PredictorReady
  - lastTransitionTime: "2024-09-30T13:11:57Z"
    message: Configuration "sklearn-v2-iris-predictor" is waiting for a Revision to
      become ready.
    reason: RevisionMissing
    severity: Info
    status: Unknown
    type: PredictorRouteReady
  - lastTransitionTime: "2024-09-30T13:11:57Z"
    message: Configuration "sklearn-v2-iris-predictor" is waiting for a Revision to
      become ready.
    reason: RevisionMissing
    status: Unknown
    type: Ready
  - lastTransitionTime: "2024-09-30T13:11:57Z"
    reason: PredictorRouteReady not ready
    severity: Info
    status: Unknown
    type: RoutesReady
  modelStatus:
    states:
      activeModelState: ""
      targetModelState: Pending
    transitionStatus: InProgress
  observedGeneration: 1
```
The deployment remains `progressing` due to the following error happening in `kserve-container`

```
2024-09-30 13:16:25.945 1 kserve INFO [storage.py:download():66] Copying contents of /mnt/models to local
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/sklearnserver/sklearnserver/__main__.py", line 35, in <module>
    model.load()
  File "/sklearnserver/sklearnserver/model.py", line 52, in load
    self._model = joblib.load(model_files[0])
  File "/prod_venv/lib/python3.9/site-packages/joblib/numpy_pickle.py", line 658, in load
    obj = _unpickle(fobj, filename, mmap_mode)
  File "/prod_venv/lib/python3.9/site-packages/joblib/numpy_pickle.py", line 577, in _unpickle
    obj = unpickler.load()
  File "/usr/local/lib/python3.9/pickle.py", line 1212, in load
    dispatch[key[0]](self)
  File "/usr/local/lib/python3.9/pickle.py", line 1537, in load_stack_global
    self.append(self.find_class(module, name))
  File "/usr/local/lib/python3.9/pickle.py", line 1579, in find_class
    __import__(module, level=0)
ModuleNotFoundError: No module named 'numpy._core'
```
**Anything else you would like to add:**
I first replicated this issue on my local system by following [serving-model-locally](https://kserve.github.io/website/latest/modelserving/v1beta1/sklearn/v2/#serving-model-locally)
I tried using ` poetry add numpy `  but since the pyproject.toml of `kserve` package has `numpy = "^1.23.5"` this failed with the following error : 
```
Using version ^2.1.1 for numpy

Updating dependencies
Resolving dependencies... (0.0s)

Because sklearnserver depends on kserve[storage] (0.14.0rc0) @ file:///Users/divyanshu/Desktop/kserve/python/kserve which depends on numpy (^1.23.5), numpy is required.
So, because sklearnserver depends on numpy (^2.1.1), version solving failed.
```
Then I tried with `poetry add numpy=1.26.4  ` but since the `poetry.lock` of `sklearnserver` package has 
```
[[package]]
name = "numpy"
version = "1.24.4"
```
this failed with the following error : 
``` 
poetry add numpy=1.26.4                                                                                                                                         ─╯

Updating dependencies
Resolving dependencies... (0.0s)

The current project's supported Python range (>=3.8,<3.12) is not compatible with some of the required packages Python requirement:
  - numpy requires Python >=3.9, so it will not be satisfied for Python >=3.8,<3.9

Because sklearnserver depends on numpy (1.26.4) which requires Python >=3.9, version solving failed.

  • Check your dependencies Python requirement: The Python requirement can be specified via the `python` or `markers` properties
    
    For numpy, a possible solution would be to set the `python` property to ">=3.9,<3.12"

    https://python-poetry.org/docs/dependency-specification/#python-restricted-dependencies,
    https://python-poetry.org/docs/dependency-specification/#using-environment-markers
```
Finally I resolved the issue locally by updating numpy version using `pip3 install numpy==1.26.4 ` , after which the `python3 sklearnserver --model_dir /path/to/model_dir --model_name sklearn-v2-iris` ran successfully.

How can I achieve this updation of numpy version or for that matter any extra dependency for K8s based deployment, which can also be personally maintainble(able to git track the changes & not have to maintain a fork of this repo, something like helm charts values.yaml) ?

Also can(&how) do I open a PR to fix this numpy version issue since it breaks the go-to sklearn deployment guide of kserve ?
 
**Environment:**

- Istio Version: 1.23.1
- Knative Version: 1.15.2
- KServe Version: 0.13
- Kubeflow version: - 
- Cloud Environment:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]: k8_istio
- Minikube/Kind version: - 
- Kubernetes version: (use `kubectl version`): v1.31.1
- OS (e.g. from `/etc/os-release`): -

**Comment by sivanantha321** [comment]
Should be fixed by https://github.com/kserve/kserve/pull/3645

**Comment by 1skol1** [comment]
@sivanantha321 any idea by when [3645](https://github.com/kserve/kserve/pull/3645) changes will be reflected in kserve realease ?

**Comment by spolti** [comment]
it will probably land in the 0.14.0 in a few weeks.

**Comment by sivanantha321** [comment]
@1skol1 It is already merged and will be available in 0.14 :smiley: .

**Comment by min731** [comment]
I was using the corresponding version of kserve 0.13.1 of sklearn 'ClusterServingRuntime', and I encountered this problem, is there any good solution other than upgrading to 0.14?

If not, I'm trying to fix it by customizing and building the sklearn 'ClusterServingRuntime'.
https://github.com/kserve/kserve/discussions/4058
